cl/citations/tasks.py

```
@@ -88,120 +102,123 @@ def find_citations_and_parentheticals_for_opinion_by_pks(
-        parentheticals: List[Parenthetical] = []
-        for _opinion, _citations in citation_resolutions.items():
-            # Currently, eyecite has a bug where parallel citations are
-            # detected individually. We avoid creating duplicate parentheticals
-            # because of that by keeping track of what we've seen so far.
-            parenthetical_texts = set()
-            for _cit in _citations:
-                # If the citation has a descriptive parenthetical, clean
-                # it up and store it as a Parenthetical
-                if (
-                    (par_text := _cit.metadata.parenthetical)
-                    and par_text not in parenthetical_texts
-                    and is_parenthetical_descriptive(par_text)
-                ):
-                    clusters_to_update_par_groups_for.add(_opinion.cluster_id)
-                    parenthetical_texts.add(par_text)
-                    clean = clean_parenthetical_text(par_text)
-                    parentheticals.append(
-                        Parenthetical(
-                            describing_opinion_id=opinion.pk,
-                            described_opinion_id=_opinion.pk,
-                            text=clean,
-                            score=parenthetical_score(clean, opinion.cluster),
-                        )
+
+def store_opinion_citations_and_update_parentheticals(
+    opinion: Opinion, index: bool
+):
+    get_and_clean_opinion_text(opinion)
+    citations: List[CitationBase] = get_citations(opinion.cleaned_text)
+
+    # If no citations are found, then there is nothing else to do for now.
+    if not citations:
+        return
+
+    # try:
+    citation_resolutions: Dict[
+        MatchedResourceType, List[SupportedCitationType]
+    ] = do_resolve_citations(citations, opinion)
+    # except ResponseNotReady as e:
+    #     # Threading problem in httplib, which is used in the Solr query.
+    #     raise self.retry(exc=e, countdown=2)
+
```

We want to keep this retry functionality in place, right? So we have a couple options:

1. Use the try/except here and return a special value in the except clause that allows the calling function to do a retry (this feels ugly).
2. Let it crash here and catch it in a try/except up in the calling function that can do a retry up there.
3. Probably something else, but those are the options that come to mind. Either way, we should make sure we maintain this try/except clause somehow.

---

Sorry, this was something I made a mental note I needed to uncomment and have in place, but it slipped my mind.

Just pushed an update. Option 2 is the path of least resistance. The `self` whose `retry` method is called is the Celery task, and that doesn't exist within the context of the new separate storage function.